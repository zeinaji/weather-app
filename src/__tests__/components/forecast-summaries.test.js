import React from "react";
import { cleanup, render } from "@testing-library/react";
import ForecastSummaries from "../../components/forecast-summaries";

const forecasts = [
  {
    date: 123,
    description: "date1",
    icon: "800",
    temperature: {
      max: 999,
    },
  },
  {
    date: 456,
    description: "date2",
    icon: "800",
    temperature: {
      max: 777,
    },
  },
];

afterEach(cleanup);

describe("ForecastSummaries", () => {
  it("renders correctly to match ForecastSummaries snapshot", () => {
    const { asFragment } = render(<ForecastSummaries forecasts={forecasts} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct amount of ForecastSummary component props", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries forecasts={forecasts} />
    );
    expect(getAllByTestId("date-id").length).toBe(2);
    expect(getAllByTestId("description-id")).toHaveLength(2);
    expect(getAllByTestId("800")).toHaveLength(2);
    expect(getAllByTestId("temperature-id")).toHaveLength(2);
  });
});
