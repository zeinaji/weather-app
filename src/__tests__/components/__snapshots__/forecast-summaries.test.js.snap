// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly to match ForecastSummaries snapshot 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      style="padding: 5px 5px 5px 5px;"
    >
      <div
        class="date"
        data-testid="date-id"
      >
        <span>
          Thu 1st Jan
        </span>
      </div>
      <div
        class="icon"
        data-testid="800"
      >
        <span>
          <i
            class="wi wi-day-sunny"
            name="owm"
          />
        </span>
      </div>
      <div
        class="temperature"
        data-testid="temperature-id"
      >
        <span>
          999°c
        </span>
      </div>
      <div
        class="description"
        data-testid="description-id"
      >
        <span>
          date1
        </span>
      </div>
      <button
        style="font-family: Balsamiq Sans, cursive; border-radius: 5px; border: 0.5px solid rgb(53, 51, 51); cursor: pointer; color: rgb(53, 51, 51);"
      >
        More Details
      </button>
    </div>
    <div
      class="forecast-summary"
      style="padding: 5px 5px 5px 5px;"
    >
      <div
        class="date"
        data-testid="date-id"
      >
        <span>
          Thu 1st Jan
        </span>
      </div>
      <div
        class="icon"
        data-testid="800"
      >
        <span>
          <i
            class="wi wi-day-sunny"
            name="owm"
          />
        </span>
      </div>
      <div
        class="temperature"
        data-testid="temperature-id"
      >
        <span>
          777°c
        </span>
      </div>
      <div
        class="description"
        data-testid="description-id"
      >
        <span>
          date2
        </span>
      </div>
      <button
        style="font-family: Balsamiq Sans, cursive; border-radius: 5px; border: 0.5px solid rgb(53, 51, 51); cursor: pointer; color: rgb(53, 51, 51);"
      >
        More Details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
